src/main/kotlin/no/nav/familie/ef/sak/arena/ArenaPeriodeUtil.kt

```diff
@@ -0,0 +1,81 @@
+package no.nav.familie.ef.sak.arena
+
+import no.nav.familie.ef.sak.repository.domain.AndelTilkjentYtelse
+import no.nav.familie.ef.sak.util.isEqualOrAfter
+import no.nav.familie.ef.sak.util.isEqualOrBefore
+import no.nav.familie.kontrakter.ef.infotrygd.InfotrygdPeriodeOvergangsstønad
+import no.nav.familie.kontrakter.ef.infotrygd.InfotrygdPerioderOvergangsstønadResponse
+import no.nav.familie.kontrakter.felles.ef.PeriodeOvergangsstønad
+import no.nav.familie.kontrakter.felles.ef.PeriodeOvergangsstønad.Datakilde
+import java.time.LocalDate
+import java.time.YearMonth
+import java.util.Stack
+
+object ArenaPeriodeUtil {
+
+    /**
+     * Skal filtrere bort de som har beløp = 0
+     * Skal filtere bort de som har tomdato < fomDato || opphørdato < tomDato
+     */
+    fun mapOgFiltrer(infotrygdPerioder: InfotrygdPerioderOvergangsstønadResponse) =
+            infotrygdPerioder.perioder.filter { it.beløp > 0 }.map {
+                PeriodeOvergangsstønad(personIdent = it.personIdent,
+                                       fomDato = it.fomDato,
+                                       tomDato = it.opphørsdatoEllerTomDato(),
+                                       datakilde = Datakilde.INFOTRYGD)
+            }.filterNot { it.tomDato.isBefore(it.fomDato) }
+
+    fun mapOgFiltrer(andeler: List<AndelTilkjentYtelse>) =
+            andeler.filter { it.beløp > 0 }.map {
+                PeriodeOvergangsstønad(personIdent = it.personIdent,
+                                       fomDato = it.stønadFom,
+                                       tomDato = it.stønadTom,
+                                       datakilde = Datakilde.EF)
+            }
+
+    private fun InfotrygdPeriodeOvergangsstønad.opphørsdatoEllerTomDato(): LocalDate {
+        val opphørsdato = this.opphørsdato
+        return if (opphørsdato != null && opphørsdato.isBefore(tomDato)) {
+            opphørsdato
+        } else {
+            tomDato
+        }
+    }
+
+    /**
+     * Slår sammen perioder som er sammenhengende og overlappende.
+     * Dette er noe som idag gjøres i infotrygd men er ikke sikkert burde gjøres når vi henter perioder fra vår egen database
+     */
+    fun slåSammenPerioder(perioder: List<PeriodeOvergangsstønad>): List<PeriodeOvergangsstønad> {
+        val mergedePerioder = Stack<PeriodeOvergangsstønad>()
+        perioder.sortedBy { it.fomDato }.forEach { period ->
+            if (mergedePerioder.isEmpty()) {
+                mergedePerioder.push(period)
+            }
+            val last = mergedePerioder.peek()
+            if (erSammenhengendeEllerOverlappende(last, period)) {
+                mergedePerioder.push(mergedePerioder.pop().copy(fomDato = minOf(last.fomDato, period.fomDato),
+                                                                tomDato = maxOf(last.tomDato, period.tomDato)))
```

Comment:
Har ikke beløp noe å si for disse mergede periodene? 🤔 Det bør kanskje alltid være "samme" beløp?

Reply:
Disse periodene (`PeriodeOvergangsstønad`) inneholder ikke beløp.
De ønsker bare sammensatte perioder, uten noe forhold til beløp :/

Reply:
Aight!